core/src/main/scala/org/apache/spark/util/SizeEstimator.scala

```diff
@@ -97,6 +98,11 @@ private[spark] object SizeEstimator extends Logging {
       return System.getProperty("spark.test.useCompressedOops").toBoolean
     }
 
+    // java.vm.info provides compressed ref info for IBM JDKs
+    if (System.getProperty("java.vendor").contains("IBM")) {
```

Comment:
I think the duplication of the snippet in launcher may be unavoidable for now. I think this PR can be left as is, and a refactoring can be added when it becomes necessary in another PR?